test/functional/eval/msgpack_functions_spec.lua

```diff
@@ -351,7 +352,8 @@ describe('msgpack*() functions', function()
     eq(1, eval('type(parsed[0]) == type(0) ' ..
                '|| parsed[0]._TYPE is v:msgpack_types.integer'))
     if eval('type(parsed[0]) == type(0)') == 1 then
-      eq(1, eval('-0x8000000000000000 == parsed[0]'))
+      command('call assert_equal(-0x8000000000000000, parsed[0])')
```

Comment:
Better use `string()` and `eq()`, it is highly uncommon in non-legacy lua tests to use parts of Vim own testing infrastructure.